test/lib/development-sandbox.ts

```diff
@@ -32,132 +32,148 @@ export function waitForHydration(browser: BrowserInterface): Promise<void> {
+        },
+        async getRedboxErrorLink() {
+          return getRedboxErrorLink(browser)
+        },
+        async getRedboxSource(includeHeader = false) {
+          const header = includeHeader ? await getRedboxHeader(browser) : ''
+          const source = await getRedboxSource(browser)
 
-        // Slow down tests a bit (we don't know how long re-rendering takes):
-        await waitFor(750)
-        return true
-      },
-      async remove(filename) {
-        await next.deleteFile(filename)
-      },
-      async evaluate(snippet: string | Function) {
-        if (typeof snippet === 'function' || typeof snippet === 'string') {
-          const result = await browser.eval(snippet)
-          await waitFor(30)
-          return result
-        } else {
-          throw new Error(
-            `You must pass a string or function to be evaluated in the browser.`
-          )
-        }
-      },
-      async assertHasRedbox() {
-        return assertHasRedbox(browser)
-      },
-      async assertNoRedbox() {
-        return assertNoRedbox(browser)
-      },
-      async waitForAndOpenRuntimeError() {
-        return waitForAndOpenRuntimeError(browser)
-      },
-      async hasErrorToast() {
-        return Boolean(await hasErrorToast(browser))
-      },
-      async getRedboxDescription() {
-        return getRedboxDescription(browser)
-      },
-      async getRedboxDescriptionWarning() {
-        return getRedboxDescriptionWarning(browser)
-      },
-      async getRedboxErrorLink() {
-        return getRedboxErrorLink(browser)
+          if (includeHeader) {
+            return `${header}\n\n${source}`
+          }
+          return source
+        },
+        async getRedboxComponentStack() {
+          return getRedboxComponentStack(browser)
+        },
+        async toggleCollapseComponentStack() {
+          return toggleCollapseComponentStack(browser)
+        },
+        async getVersionCheckerText() {
+          return getVersionCheckerText(browser)
+        },
       },
-      async getRedboxSource(includeHeader = false) {
-        const header = includeHeader ? await getRedboxHeader(browser) : ''
-        const source = await getRedboxSource(browser)
-
-        if (includeHeader) {
-          return `${header}\n\n${source}`
+      get [Symbol.asyncDispose]() {
+        unwrappedByTypeScriptUsingKeyword = true
+        return async () => {
+          await browser.close()
+          await next.stop()
+          await next.clean()
         }
-        return source
-      },
-      async getRedboxComponentStack() {
-        return getRedboxComponentStack(browser)
-      },
-      async toggleCollapseComponentStack() {
-        return toggleCollapseComponentStack(browser)
       },
-      async getVersionCheckerText() {
-        return getVersionCheckerText(browser)
-      },
-    },
-    async cleanup() {
-      await browser.close()
-      await next.stop()
-      await next.clean()
-    },
+    }
+  } finally {
```

Comment:
should we also have a `catch` here to log the errors in case sth wrong?

Reply:
The exception will still bubble up and eventually fail the test as intended.